src/server/scene/surface_creation_parameters.cpp

```diff
@@ -250,7 +252,11 @@ bool ms::operator==(
         lhs.confine_pointer == rhs.confine_pointer &&
         lhs.depth_layer == rhs.depth_layer &&
         lhs.attached_edges == rhs.attached_edges &&
-        lhs.exclusive_rect == rhs.exclusive_rect;
+        lhs.exclusive_rect == rhs.exclusive_rect &&
+        lhs.depth_layer == rhs.depth_layer &&
+        lhs.attached_edges == rhs.attached_edges &&
+        lhs.exclusive_rect == rhs.exclusive_rect &&
```

Comment:
Looks like duplication of depth_layer, attached_edges and exclusive_rect checks.